vmm/src/config.rs

```diff
@@ -2144,6 +2150,16 @@ impl VmConfig {
             }
         }
 
+        #[cfg(feature = "sev_snp")]
+        {
+            let host_data_opt = &self.payload.as_ref().unwrap().host_data;
+
+            if let Some(host_data) = host_data_opt {
+                if !host_data.is_empty() && host_data.len() != 64 {
```

Comment:
What's the benefit of supporting empty `host_data`? Why not simply require users not to provide the "--host-data" option at all? It will make your validation and processing of `host_data` cleaner.

Reply:
@likebreath  Thank you. I have addressed all of your comments.